src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -19,13 +19,14 @@ namespace System.ServiceProcess
     /// and manipulate it or get information about it.
     public class ServiceController : Component
     {
-        private string _machineName;
+        private string _machineName = DefaultMachineName; // Never null, always a valid name
         private readonly ManualResetEvent _waitForStatusSignal = new ManualResetEvent(false);
         private const string DefaultMachineName = ".";
 
-        private string _name;
-        private string _eitherName;
-        private string _displayName;
+        private string _name = ""; // Never null
+        private string _eitherName = ""; // Never null
+        private string _displayName = ""; // Never null
```

Comment:
This has potential to create corner case bugs when the service name is empty string. Using `null` to mean not-yet-computed looks better.